docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/data-import/spark-writer.md

```diff
@@ -0,0 +1,344 @@
+mvn compile package
+```
+
+### 2.2 oss Download
+
+```bash
+wget https://nebula-graph.oss-cn-hangzhou.aliyuncs.com/jar-packages/sst.generator-1.0.0-beta.jar
+```
+
+## 3. User Guide
+
+This section includes the following steps:
+
+* Create a graph space and its schema in Nebula
+* Write data files
+* Write input source mapping file
+* Import data
+
+### 3.1 Create Graph Space
+
+Please refer to the example graph in [Quick Start](../../../../../1.overview/2.quick-start/1.get-started.md)。
+
+### 3.2 Example Data
+
+#### 3.2.1 Vertices
+
+A vertex data file consists of multiple rows. Generally one row represents one vertex, and one of the columns is the ID of the vertex. This ID column is specified in the mapping file. Other columns are the properties of the vertex. Consider the following example in JSON format.
+
+* **Player** data
+
+```text
+{"id":100,"name":"Tim Duncan","age":42}
+{"id":101,"name":"Tony Parker","age":36}
+{"id":102,"name":"LaMarcus Aldridge","age":33}
+```
+
+#### 3.2.2 Edges
+
+An edge data file consists of multiple rows. Generally one row represents one edge, and one of the columns is the ID of the source vertex, the other column is the ID of the dest vertex. These ID columns are specified in the mapping file. Other columns are the properties of the edge. Consider the following example in JSON format.
+
+* Edge without rank, take edge _**follow**_ as example
+
+```text
+{"source":100,"target":101,"likeness":95}
+{"source":101,"target":100,"likeness":95}
+{"source":101,"target":102,"likeness":90}
+```
+
+* Edge with rank, take edge _**follow**_ as example
+
+```text
+{"source":100,"target":101,"likeness":95,"ranking":2}
+{"source":101,"target":100,"likeness":95,"ranking":1}
+{"source":101,"target":102,"likeness":90,"ranking":3}
+```
+
+#### 3.2.3 Supporting for Geo Data
+
+Spark Writer supports importing Geo data. Geo data contains latitude and longitude, and the data type is double.
+
+```text
+{"latitude":30.2822095,"longitude":120.0298785,"target":0,"dp_poi_name":"0"}
+{"latitude":30.2813834,"longitude":120.0208692,"target":1,"dp_poi_name":"1"}
+{"latitude":30.2807347,"longitude":120.0181162,"target":2,"dp_poi_name":"2"}
+{"latitude":30.2812694,"longitude":120.0164896,"target":3,"dp_poi_name":"3"}
+```
+
+#### 3.2.4 Data Source Files
+
+The currently supported data sources by Spark Writer are:
+
+* HDFS
+* HIVE
+
+##### HDFS Files
+
+HDFS supports the following file formats:
+
+* Parquet
+* JSON
+* CSV
+* ORC
+
+Player data in Parquet format:
+
+```text
++-------+---+---------+
+|age| id|        name|
++-------+---+---------+
+| 42|100| Tim Duncan |
+| 36|101| Tony Parker|
++-------+---+---------+
+```
+
+In JSON:
+
+```text
+{"id":100,"name":"Tim Duncan","age":42}
+{"id":101,"name":"Tony Parker","age":36}
+```
+
+In CSV:
+
+```text
+age,id,name
+42,100,Tim Duncan
+36,101,Tony Parker
+```
+
+##### Database
+
+Spark Writer supports database as the data source, only HIVE is available for now.
```

Comment:
Spark Writer supports database as the data source, and only HIVE is available now.